src/server/scripts/KhazAlgar/CityOfThreads/boss_orator_krix_vizk.cpp

```diff
@@ -0,0 +1,431 @@
+    }
+
+    void Tick(AuraEffect const* /*aurEff*/)
+    {
+        GetCaster()->CastSpell(GetTarget(), SPELL_VOCIFEROUS_INDOCTRINATION_DAMAGE, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .OriginalCastId = GetAura()->GetCastId()
+        });
+    }
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        GetTarget()->CastSpell(GetCaster(), SPELL_LINGERING_INFLUENCE_AREATRIGGER, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR
+        });
+
+        if (Creature* creatureTarget = GetTarget()->ToCreature())
+            creatureTarget->SetPower(POWER_ENERGY, 0);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_orator_krix_vizk_vociferous_indoctrination_periodic::Tick, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+        AfterEffectRemove += AuraEffectRemoveFn(spell_orator_krix_vizk_vociferous_indoctrination_periodic::OnRemove, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 434691 - Chains of Oppression
+// Id - 31997
+struct at_orator_krix_vizk_chains_of_oppression : AreaTriggerAI
+{
+    at_orator_krix_vizk_chains_of_oppression(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (!unit->IsPlayer())
+            return;
+
+        at->GetCaster()->CastSpell(unit, SPELL_CHAINS_OF_OPPRESSION_PERIODIC, TRIGGERED_FULL_MASK);
+    }
+
+    void OnUnitExit(Unit* unit) override
+    {
+        if (!unit->IsPlayer())
+            return;
+
+        unit->RemoveAurasDueToSpell(SPELL_CHAINS_OF_OPPRESSION_PERIODIC);
+    }
+};
+
+// 434923 - Lingering Influence
+// ID - 32026
+struct at_orator_krix_vizk_lingering_influence : AreaTriggerAI
+{
+    at_orator_krix_vizk_lingering_influence(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (!unit->IsPlayer())
+            return;
+
+        at->GetCaster()->CastSpell(unit, SPELL_LINGERING_INFLUENCE_DAMAGE, TRIGGERED_FULL_MASK);
+    }
+
+    void OnUnitExit(Unit* unit) override
+    {
+        if (!unit->IsPlayer())
+            return;
+
+        unit->RemoveAurasDueToSpell(SPELL_LINGERING_INFLUENCE_DAMAGE);
+    }
+
+    void OnInitialize() override
+    {
+        std::array<DBCPosition2D, 2> points =
+        { {
+            { 0.0f, 1.0f },
+            { 1.0f, 18.0f }
+        } };
+
+        at->SetTimeToTargetScale(6000);
+        at->SetOverrideScaleCurve(points, 2);
+    }
+};
+
+void AddSC_boss_orator_krix_vizk()
+{
+    new GenericAreaTriggerEntityScript<at_orator_conversation_intro<CONVERSATION_ORATOR_INTRO_1, DATA_ORATOR_CONVO>>("at_orator_conversation_intro_1");
```

Comment:
the 2nd parameter here `DATA_ORATOR_CONVO` is used for `ObjectAccessor::GetCreature()` i assume you want `DATA_ORATOR_KRIX_VIZK` here aswell. Given that its equal for all 3 cases then you can even remove that parameter and use it directly in the `at_orator_conversation_intro` script

Reply:
not equal, one of convos are done by another orator npc (different entry)